### PR TITLE
feat: remove ui-kit

### PR DESCRIPTION
**Pull request by mallachari**
Fixes: https://github.com/stoplightio/elements/issues/957

- replaces all ui-kit components with mosaic ones
- replaces classes with mosaic components
- adds `sl` prefixes to all tailwind classes
- removes ui-kit styles
- removes ui-kit dependencies

~~For some reason `Tooltip` and `Popover` do not appear on hover or on click - I'll debug it, but we use it rarely and it's hardly noticeable so if that still causes problems we could handle it separately.~~
`Tooltip` doesn't work for badge currently as it doesn't support hover - it will be fixed in mosaic.

### CODE REVIEW

File: packages/elements-core/src/components/Docs/HttpOperation/Badges.tsx
Comment (marcelltoth) [inline packages/elements-core/src/components/Docs/HttpOperation/Badges.tsx:39]: `badge` is not a valid role. [See spec](https://www.w3.org/TR/wai-aria-1.1/#roles). I know it has been here, I also mentioned it before, but now that you are changing tests anyway we could clean this up too.

File: packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.spec.tsx
Comment (marcelltoth) [inline packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.spec.tsx:581]: Why does this need to be so complicated? Won't the original do? If it's a mosaic accessibility problem, can we fix it?

File: packages/elements-core/src/components/Docs/Skeleton.tsx
Comment (marcelltoth) [inline packages/elements-core/src/components/Docs/Skeleton.tsx:10]: These are still using unprefixed classnames here

File: packages/elements/src/components/API/APIWithStackedLayout.tsx
Comment (marcelltoth) [inline packages/elements/src/components/API/APIWithStackedLayout.tsx:162]: If you decided to add a `Box` you could might as well just use it's `display={}` prop, couldn't you?

File: packages/elements-core/src/components/NonIdealState.tsx
Comment (marcelltoth) [inline packages/elements-core/src/components/NonIdealState.tsx:9]: Child or Node? 😉 IDK which one's better but let's be consistent.

**Review by marcelltoth — COMMENTED** [review]
Gotta love the branch name 😂

File: packages/elements-core/src/components/Docs/Article/Headings.tsx
Comment (mallachari) [reply packages/elements-core/src/components/Docs/Article/Headings.tsx:56]: Yup, exactly that :)

File: packages/elements-core/src/components/Docs/Article/Headings.tsx
Comment (mallachari) [reply packages/elements-core/src/components/Docs/Article/Headings.tsx:45]: Well, it should appear at the bottom. I guess the top right was just flipped to the bottom anyway because of lack of space

File: packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.spec.tsx
Comment (mallachari) [reply packages/elements-core/src/components/Docs/HttpOperation/HttpOperation.spec.tsx:581]: Because the `badge` role is on the parent div and it `findByRole` won't find the text content. That was the solution I found working, though perhaps we could access it different way

File: packages/elements-core/src/components/Docs/Skeleton.tsx
Comment (mallachari) [reply packages/elements-core/src/components/Docs/Skeleton.tsx:10]: Good point!

File: packages/elements-core/src/components/NonIdealState.tsx
Comment (mallachari) [reply packages/elements-core/src/components/NonIdealState.tsx:9]: That comes from original ui-kit component, I just used the same props. But yes we can have it unified

**Comment by mmiask** [comment]
@mallachari The stacked layout takes a bit long to load in Storybook, but the most important thing is that its styling is off. Let's make sure we address that before merging

Screenshot 2021-05-31 at 17 12 04

**Comment by mallachari** [comment]
> @mallachari The stacked layout takes a bit long to load in Storybook, but the most important thing is that its styling is off. Let's make sure we address that before merging
> 
> Screenshot 2021-05-31 at 17 12 04

Hmm, I've tested it and it was fine few commits back. I'll check it, thanks!

**Review by mmiask — COMMENTED** [review]
I am very impressed with how this turned out 👏 I don't have any major concerns or comments besides the styling in a stacked layout, that I mentioned in another comment.

**Comment by philsturgeon** [comment]
All good from me when you've got technical problems figured out.

**Comment by philsturgeon** [comment]
Stoplight-Flavoured Markdown still needs some love. 

**Before**

Screen Shot 2021-06-02 at 2 36 58 PM

**After**

Screen Shot 2021-06-02 at 2 36 50 PM

**Review by philsturgeon — CHANGES_REQUESTED** [review]

File: packages/elements-core/src/components/Docs/HttpService/HttpService.tsx
Comment (mallachari) [reply packages/elements-core/src/components/Docs/HttpService/HttpService.tsx]: It is moved out to constants file, but we need to use something like that here as Badge component comes with very limited set of colors: `IntentVals = 'default' | 'success' | 'warning' | 'danger';`

**Comment by mallachari** [comment]
All the issues are solved. Markdown viewer is updated and ui-kit is no more.

**Comment by mmiask** [comment]
### ⚠️ Font size differ between sections. As much as it looks good for the article, it is a bit odd for me in the `Security` panel

Screenshot 2021-06-08 at 12 45 18

### ⚠️ Same thing in stacked layout - can we make it more consistent?

Screenshot 2021-06-08 at 12 48 45

### ⚠️ We are duplicating the header content in the description

Screenshot 2021-06-08 at 12 50 25

### ⚠️ Dark mode in stacked layout is not working properly:

Screenshot 2021-06-08 at 12 52 18

### ⚠️ Error screens in dark mode are a bit hard to see

Screenshot 2021-06-08 at 12 53 28

### ❤️  This looks great, I love it 👏 

Screenshot 2021-06-08 at 12 57 21

### ❓ I don't know if this related to MarkdownViewer or styling in Elements, but the table width is different

We are now using the full width of the section instead of minimal space required for the content. @noor-ahmad - what are your thoughts on that?

**Now**:

Screenshot 2021-06-08 at 13 02 37

**Before**:
Screenshot 2021-06-08 at 13 02 15

### ⚠️ Link for that logo is outdated - not needed for this PR, but worth knowing

Screenshot 2021-06-08 at 13 04 08

### ⚠️ Bullet points with links look like plain text, don't have the links styling

Screenshot 2021-06-08 at 13 06 56

### ⚠️ Padding for schemas embedded in articles doesn't look good (the `required` text)

Screenshot 2021-06-08 at 13 09 47

**Review by mmiask — APPROVED** [review]
I'm tentatively approving this one ✅ 

Even though there are still multiple styling issues here, the code itself looks good and it does the expected thing - removes `ui-kit`, significantly reducing the amount of code.

If other folks are good with that, I'm ok with merging this one and addressing the styling issues in a separate PR.

**Comment by mallachari** [comment]
@mmiask 
1. The font size is a matter of markdown-viewer font that changed. We can set it with styling those elements.
2. The dark mode issues night be treated separately as they are unrelated to ui-kit removal
3. Title duplication shouldn't be related to any change here. Are you sure it wasn't there before?
4. Table changed because MDV changed. If we really wan't to keep it old way we could add custom styling, but I'm not sure if we want that.
5. I'll check the links but they are too brought with MDV. 
6. The padding in JSV is the same as it was before. There's just a border that appeared and I'm not sure why it did, perhaps it was brought with main form other pr I'll check it.

**Comment by mallachari** [comment]
We can't merge it anyway for now. We need to keep pr open until platform is ready, but the issue itself is done.

**Comment by mmiask** [comment]
> @mmiask
> 
> 1. The font size is a matter of markdown-viewer font that changed. We can set it with styling those elements.
> 2. The dark mode issues night be treated separately as they are unrelated to ui-kit removal
> 3. Title duplication shouldn't be related to any change here. Are you sure it wasn't there before?
> 4. Table changed because MDV changed. If we really wan't to keep it old way we could add custom styling, but I'm not sure if we want that.
> 5. I'll check the links but they are too brought with MDV.
> 6. The padding in JSV is the same as it was before. There's just a border that appeared and I'm not sure why it did, perhaps it was brought with main form other pr I'll check it.

@mallachari then I'm even more confident in merging this one as soon as we get the platform ready

**Comment by philsturgeon** [comment]
Can somebody create issues for any future work that needs doing? Maybe @mmiask you could make one genera “style tweaks issue” that covers your feedback there, and we can add other little style things we notice over the next few days.

**Comment by marbemac** [comment]
> Tooltip doesn't work for badge currently as it doesn't support hover - it will be fixed in mosaic.

As of mosaic 64 any tooltip trigger should work so long as it forwards ref.

**Comment by marbemac** [comment]
> Font size differ between sections. As much as it looks good for the article, it is a bit odd for me in the Security panel

Totally agree - by default MarkdownViewer is setup to render longform markdown content. When we use it in denser areas of the UI, we probably want to set an explicit fontSize to match the markdown sizing with the ui around it. In the case of the right panel sections, looks like font size of 12, which can be achieved in markdown viewer with something like this:

```tsx
<Box
  as={MarkdownViewer}
  markdown={value}
  style={{
    fontSize: 12,
  }}
/>
```

> I don't know if this related to MarkdownViewer or styling in Elements, but the table width is different

We opted for consistency re always full width tables.

> Bullet points with links look like plain text, don't have the links styling

This has been fixed if bump Mosaic.

---

> We can't merge it anyway for now. We need to keep pr open until platform is ready

@mallachari what's the blocking issue(s)?

**Comment by mallachari** [comment]
@marbemac platform still uses TableOfContents container that depends on ui-kit ToC so we can't merge this one until it's ready (to not lock platform with older elements version). We are currently working on https://github.com/stoplightio/platform-internal/issues/6255 which should deal with it (was blocked earlier by other issues). Once it's done I'll update this PR.

**Comment by marbemac** [comment]
Ah so if you `yalc` this branch into platform-internal, things break? `ui-kit` will still be loaded and available in platform-internal, even if elements v7 no longer uses it. But not a big deal, can wait if that's easier.

**Comment by mallachari** [comment]
> Ah so if you `yalc` this branch into platform-internal, things break? `ui-kit` will still be loaded and available in platform-internal, even if elements v7 no longer uses it. But not a big deal, can wait if that's easier.

Ui-kit is not a problem here. What would break is the component from elements that is removed in this branch but present in platform. And that's what we are replacing in stoplightio/platform-internal#6255. It's bit tricky because moving platform to new toc means having new uris (based on node slug) so we need to adjust docs to handle that when the feature flag is set

**Comment by marbemac** [comment]
AH yup! Ok got it, thanks for clarifying.

**Comment by mallachari** [comment]
@marbemac do you perhaps know why Menu component might not be working within this PR? I've merged changes from main, bumped markdown, markdown-viewer and jsv to stable versions, mosaic is in v1 but even in demo app it doesn't show it for some reason.

**Comment by mallachari** [comment]
Alright, this PR is finally free of any blockers and ready to get in. Give it one more look and we should be able to close it.

**Review by mmiask — APPROVED** [review]
Once again, great job with that @mallachari! 🤘  Code-wise it looks good, I already reviewed it once before and with the necessary PRs merged (that were blocking this), I think it's good to go! 🚀 

As for styling, I think @noor-ahmad would be the best person to review it, but IMHO it looks great already and we can make any of the potiential UI tweaks after this PR

**Pull request merged**
